bluemira/equilibria/opt_constraints.py

```diff
@@ -20,312 +20,253 @@
 # License along with bluemira; if not, see <https://www.gnu.org/licenses/>.
 
 """
-Equilibrium optimisation constraint functions.
-for use in NLOpt constrained
-optimisation problems.
-
-Constraint functions must be of the form:
-
-.. code-block:: python
-
-    def f_constraint(constraint, x, grad, args):
-        constraint[:] = my_constraint_calc(x)
-        if grad.size > 0:
-            grad[:] = my_gradient_calc(x)
-        return constraint
-
-The constraint function convention is such that c <= 0 is sought. I.e. all constraint
-values must be negative.
-
-Note that the gradient (Jacobian) of the constraint function is of the form:
-
-.. math::
-
-    \\nabla \\mathbf{c} = \\begin{bmatrix}
-            \\dfrac{\\partial c_{0}}{\\partial x_0} & \\dfrac{\\partial c_{0}}{\\partial x_1} & ... \n
-            \\dfrac{\\partial c_{1}}{\\partial x_0} & \\dfrac{\\partial c_{1}}{\\partial x_1} & ... \n
-            ... & ... & ... \n
-            \\end{bmatrix}
-
-The grad and constraint matrices must be assigned in place.
-
-If grad is not updated, the constraint can still be used for derivative-free
-optimisaiton algorithms, but will need to be updated or approximated for use
-in derivative based algorithms, such as those utilising gradient descent.
+Equilibrium optimisation constraint classes
 """  # noqa (W505)
 
 from abc import ABC, abstractmethod
+from copy import deepcopy
 from typing import List, Union
 
 import numpy as np
 
+from bluemira.equilibria.opt_constraint_funcs import (
+    Ax_b_constraint,
+    L2_norm_constraint,
+    coil_field_constraints,
+    coil_force_constraints,
+)
 from bluemira.equilibria.plotting import ConstraintPlotter
 from bluemira.geometry._deprecated_loop import Loop
 from bluemira.utilities.opt_problems import OptimisationConstraint
 from bluemira.utilities.tools import abs_rel_difference, is_num
 
 
-def objective_constraint(constraint, vector, grad, objective_function, maximum_fom=1.0):
+def _get_dummy_equilibrium(equilibrium, I_not_dI):
     """
-    Constraint function to constrain the maximum value of an NLOpt objective
-    function provided
+    Get a dummy equilibrium for current optimisation where the background response is
+    solely due to the plasma and passive coils.
 
-    Parameters
-    ----------
-    objective_function: callable
-        NLOpt objective function to use in constraint.
-    maximum_fom: float (default=1.0)
-        Value to constrain the objective function by during optimisation.
+    Notes
+    -----
+    When we do dI (current gradient) optimisation, the background vector includes the
+    contributions from the whole coilset (including active coils).
+
+    When we do I (current vector) optimisation, the background vector only includes
+    contributions from the passive coils (plasma).
     """
-    constraint[:] = objective_function(vector, grad) - maximum_fom
-    return constraint
+    if I_not_dI:
```

Comment:
As before - do we need this if statement? Seems easier to just call `_get_dummy_equilibrium` when we need to and not need I_not_dI in this function at all.